tests/OutputTest/OutputTest.php

```diff
@@ -0,0 +1,209 @@
+<?php
+namespace Dompdf\Tests\OutputTest;
+
+use CallbackFilterIterator;
+use Dompdf\Tests\TestCase;
+use FilesystemIterator;
+use Iterator;
+use PHPUnit\Framework\AssertionFailedError;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use RuntimeException;
+use SplFileInfo;
+use Symfony\Component\Process\Process;
+
+final class OutputTest extends TestCase
+{
+    private const DATASET_DIRECTORY = __DIR__ . "/../_files/OutputTest";
+    private const FAILED_OUTPUT_DIRECTORY = __DIR__ . "/../../tmp/failed-output-tests";
+
+    private static function datasetName(SplFileInfo $file): string
+    {
+        $prefixLength = strlen(self::DATASET_DIRECTORY);
+        $path = substr($file->getPath(), $prefixLength + 1);
+        $name = $file->getBasename("." . $file->getExtension());
+        return "$path/$name";
+    }
+
+    /**
+     * @return Iterator<Dataset>
+     */
+    public static function datasets(): Iterator
+    {
+        $flags = FilesystemIterator::KEY_AS_FILENAME
+            | FilesystemIterator::CURRENT_AS_FILEINFO
+            | FilesystemIterator::SKIP_DOTS;
+        $filter = function (SplFileInfo $file) {
+            return $file->getExtension() === "html";
+        };
+        $dir = new RecursiveDirectoryIterator(self::DATASET_DIRECTORY, $flags);
+        $files = new CallbackFilterIterator(new RecursiveIteratorIterator($dir), $filter);
+
+        foreach ($files as $file) {
+            $name = self::datasetName($file);
+            yield new Dataset($name, $file);
+        }
+    }
+
+    public static function outputTestProvider(): Iterator
+    {
+        foreach (self::datasets() as $dataset) {
+            yield $dataset->name => [$dataset];
+        }
+    }
+
+    protected function setUp(): void
+    {
+        $process = new Process(["gs", "-v"]);
+        $exitCode = $process->run();
+
+        if ($exitCode === 127) {
+            $this->markTestSkipped("Output tests need Ghostscript to be installed");
```

Comment:
We could suggest installing `sudo apt-get install ghostscript`
I dont know how we have setup phpunit for skipping test but i think it should definitely fail. is `stopOnSkipped="true"`?

Reply:
> We could suggest installing `sudo apt-get install ghostscript`

Added a short note.

> I dont know how we have setup phpunit for skipping test but i think it should definitely fail. is `stopOnSkipped="true"`?

I think we want to have `failOnSkipped="true"`? CI should fail without the output tests, but locally you might want to run the testsuite without Ghostscript, and still have results for all the other tests displayed.